lib/spack/spack/package.py

```diff
@@ -673,17 +681,40 @@ def url_for_version(self, version):
             version = Version(version)
 
         cls = self.__class__
-        if not (hasattr(cls, 'url') or self.version_urls()):
-            raise NoURLError(cls)
-
-        # If we have a specific URL for this version, don't extrapolate.
-        version_urls = self.version_urls()
-        if version in version_urls:
-            return version_urls[version]
 
-        # If we have no idea, try to substitute the version.
-        return spack.url.substitute_version(
-            self.nearest_url(version), self.url_version(version))
+        if hasattr(cls, 'url') or self.version_urls():
+            # If we have a specific URL for this version, don't extrapolate.
+            version_urls = self.version_urls()
+            if version in version_urls:
+                return version_urls[version]
+
+            # If we have no idea, try to substitute the version.
+            return spack.url.substitute_version(
+                self.nearest_url(version), self.url_version(version))
+        elif hasattr(cls, 'pypi') or hasattr(self, 'pypi'):
+            client = xmlrpclib.ServerProxy('https://pypi.python.org/pypi')
```

Comment:
So the UI is to do:
```
class MyPackage(Package):
    url='http://...'

class MyPyPIPackage(Package):
    pypi='packagename'
```
Why not consider this alternative:
```
class MyPackage(Package):
    url='pypi://packagename'
```
I think we should at least consider the pros and cons of these two different UIs.